src/Backend/Core/Engine/Authentication.php

```diff
@@ -300,21 +308,28 @@ public static function isAllowedModule($module)
         if (empty(self::$allowedModules)) {
             // init var
             $db = BackendModel::get('database');
+            $em = BackendModel::get('doctrine.orm.entity_manager');
 
-            // get allowed modules
-            $allowedModules = $db->getColumn(
-                'SELECT DISTINCT grm.module
+            // get the group id
```

Comment:
You do this $groupId = $db->getVar twice. Maybe refactor it to a method "getGroupIdForCurrentUser" or something like this. This possibly already exists. Can't you do ->getGroup() on the current user object?